File: LC 847.cpp
class Solution {
public:
    int shortestPathLength(vector<vector<int>>& graph) {
        int n = graph.size();
        vector<vector<int>> vis(n, vector<int> (1 << n, 0));
        queue<tuple<int, int, int>> q;
        for (int i = 0; i < n; ++i) {
            q.emplace(i, 1 << i, 0);
            vis[i][1<<i] = 1;
        }
        int ans = 0;
        while (!q.empty()) {
            auto [u, mask, dis] = q.front();
            q.pop();
            if (mask == (1 << n) - 1) {
                ans = dis;
                break;
            }
            for (auto v : graph[u]) {
                int mask_v = mask | (1 << v);
                if (!vis[v][mask_v]) {
                    vis[v][mask_v] = 1;
                    q.push({v, mask_v, dis + 1});
                }
            }
        }
        return ans;
    }
};

File: 1000/LC 847.go
func shortestPathLength(graph [][]int) (ans int) {
    n := len(graph)
    vis := make([][]bool, n)
    for i := range vis {
        vis[i] = make([]bool, 1 << n)
    }
    type tuple struct {
        cur, mask, dis int
    }
    q := []tuple{}
    for i := range graph {
        q = append(q, tuple{i, 1 << i, 0})
        vis[i][1<<i] = true
    }

    for len(q) > 0 {
        tmp := q[0]
        q = q[1:]
        cur, mask, dis := tmp.cur, tmp.mask, tmp.dis
        if mask == (1 << n) - 1{
            ans = dis
            break
        }
        for _, x := range graph[cur] {
            mask_cur := mask | (1 << x)
            if !vis[x][mask_cur] {
                vis[x][mask_cur] = true
                q = append(q, tuple{x, mask_cur, dis + 1})
            }
        }
    }

    return
}